### PR TITLE
Load balancer client with register

### PR DESCRIPTION
**Pull request by mpetrun5**
Closes: #5

### CODE REVIEW

**Review by MakMuftic — APPROVED** [review]

File: internal/lb/client.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+// Register daemon with load balancer and store token in client
+func (c *Client) Register(id string, nodeURL string, payoutAddress string, configHash string) error {
+	body := &RegisterRequest{
+		ID:            id,
+		NodeURL:       nodeURL,
+		PayoutAddress: payoutAddress,
+		ConfigHash:    configHash,
+	}
+	req, _ := c.NewRequest(http.MethodPost, registerEndpoint, body)
```
Comment (mpetrunic) [inline internal/lb/client.go:53]: you could have malformed url error here

File: internal/lb/client.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	if tokenResponse.Token != "" {
+		c.Token = tokenResponse.Token
+	}
+
+	return err
+}
+
+// NewRequest creates an API request. A relative URL can be provided in urlStr, which will be resolved to the
+// BaseURL of the Client. If Client contains token X-Auth-Header will be added to request.
+func (c *Client) NewRequest(method string, urlStr string, body interface{}) (*http.Request, error) {
```
Comment (mpetrunic) [inline internal/lb/client.go:66]: Don't export NewRequest and Do method

**Review by mpetrunic — CHANGES_REQUESTED** [review]

**Review by MakMuftic — APPROVED** [review]

File: internal/lb/client.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+// TokenResponse from lb register endpoint
+type TokenResponse struct {
+	Token string `json:"token"`
+}
+
+// Client used to communicate with vedran load balancer
+type Client struct {
+	client  *http.Client
+	BaseURL *url.URL
```
Comment (mpetrun5) [inline internal/lb/client.go:33]: url.Parse doesn't return error for invalid url
Will need to check that differently

Comment (mpetrun5) [reply internal/lb/client.go:33]: Checked and should be fine
NewRequest still returns req and do fails as it should

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**